include/common/buf.h

```diff
@@ -55,13 +60,35 @@ void buf_add_char(struct buf *s, char data);
 /**
  * buf_clear - clear the buffer, internal allocations are preserved
  * @s: buffer
+ *
+ * The buffer will be set to a NUL-terminated empty string.
+ *
+ * This is the appropriate function to call to re-use the buffer
+ * in a loop or similar situations as it reuses the existing heap
+ * allocation.
  */
 void buf_clear(struct buf *s);
 
 /**
  * buf_reset - reset the buffer, internal allocations are free'd
  * @s: buffer
+ *
+ * The buffer will be re-initialized to BUF_INIT (empty string).
+ *
+ * Inside a loop, consider using buf_clear() instead, as it allows
+ * reusing the existing heap allocation. buf_reset() should still be
+ * called after exiting the loop.
  */
 void buf_reset(struct buf *s);
```

Comment:
While I think what `buf_reset()` is now doing is good I am not sure about using `xxx_reset()` to free resources really matches the rest of the codebase. We usually use `_destroy()` instead. Maybe we could just add another wrapper?

Reply:
In my mind `reset` suggests "free resources and set object back to initial state", i.e. it's a combined `destroy` + `init`. This has precedent outside labwc, for example (C++ again) `std::unique_ptr::reset`.

Reply:
Right, and IMHO we should keep the `buf_reset()` call. It is being forced to use it to free something that feels a bit strange to me.

Reply:
I think if the `struct buf` itself was allocated then `destroy` would make more sense. Since the `struct buf` itself is not freed, `reset` emphasizes that it's left in a valid state (doesn't contain a stale `buf` pointer etc.)

Reply:
I'm good with `reset`. For comparison `git` uses `release`. 

https://github.com/git/git/blob/8f7582d995682f785e80e344197cc715e6bc7d8e/strbuf.c#L75

Reply:
Fine.